extensions/resource-deployment/src/test/services/kubeService.test.ts

```diff
@@ -0,0 +1,79 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as fs from 'fs';
+import 'mocha';
+import * as path from 'path';
+import * as sinon from 'sinon';
+import * as yamljs from 'yamljs';
+import { tryExecuteAction } from '../../common/utils';
+import { getDefaultKubeConfigPath, getKubeConfigClusterContexts, KubeClusterContext, KubeService } from '../../services/kubeService';
+import should = require('should');
+
+const kubeConfig =
+{
+	'contexts': [
+		{
+			'context': {
+				'cluster': 'docker-desktop',
+				'user': 'docker-desktop'
+			},
+			'name': 'docker-for-desktop'
+		},
+		{
+			'context': {
+				'cluster': 'kubernetes',
+				'user': 'kubernetes-admin'
+			},
+			'name': 'kubernetes-admin@kubernetes'
+		}
+	],
+	'current-context': 'docker-for-desktop'
+};
+describe('KubeService', function (): void {
+	const kubeService: KubeService = new KubeService();
+	const configFile = 'kubeConfig';
+
+	afterEach('NotebookService cleanup', () => {
+		sinon.restore();
+	});
+
+	it('getDefaultKubeConfigPath', async () => {
+		getDefaultKubeConfigPath().should.endWith(path.join('.kube', 'config'));
+		kubeService.getDefaultConfigPath().should.endWith(path.join('.kube', 'config'));
+	});
+
+	describe('getKubeConfigClusterContexts', () => {
+		it('success', async () => {
+			sinon.stub(fs.promises, 'access').withArgs(configFile).resolves(); //resolving access to file, mocks its existence
+			sinon.stub(yamljs, 'load').returns(<any>kubeConfig);
+			const verifyConfigs = (configs: KubeClusterContext[]) => {
+				configs.length.should.equal(2);
```

Comment:
Make sure these checks indicate which one of the calls is failing - right now there's no way to tell if it was `getKubeConfigClusterContexts` or `getClusterContexts`

Reply:
The context of which test failed is enough to differentiate but happy to add message strings about the failed tests.

Reply:
done